app/controllers/ally_controller.rb

```diff
@@ -0,0 +1,142 @@
+class AllyController < ApplicationController
+
+  #
+  # REST methods
+  #
+
+  ##
+  # Returns whether the plugin is enabled for the account. If its
+  # enabled extra information such as the client ID and base URL
+  # of the Ally API is returned as well
+  def enabled
+    plugin_data = get_plugin_data
+    data = {
+      :enabled => plugin_data[:enabled],
+      :clientId => plugin_data[:client_id],
+      :baseUrl => plugin_data[:base_url]
+    }
+    render :json => data
+  end
+
+  ##
+  # Sign a request to the Ally REST API but don't execute it.
+  # The UI will execute the request
+  def sign
+    if (request = get_request())
```

Comment:
What on earth is going on here? falsey check, and returning an undefined value? Blasphemy!